compass/ocean/tests/global_ocean/init/initial_state.py

```diff
@@ -163,5 +164,9 @@ def run(self):
         add_mesh_and_init_metadata(self.outputs, config,
                                    init_filename='initial_state.nc')
 
+        particles.write(init_filename='initial_state.nc', particle_filename='particles.nc',
+                        graph_filename='graph.info.part.{}'.format(cores),
+                        types='buoyancy')
```

Comment:
I copied this particle creation line from the ziso test. I wasn't sure where to put it - here (would run every time) or within the `lagrangian_particles` test.

Reply:
I think this belong as an optional part of the `ForwardStep`.  You should add an argument to the constructor of `ForwardStep` to indicate that particles should be included.  Then, store that argument in the class (e.g. `self.with_particles = with_particles`).  In the `run()` method, you can call this function.  That will be with the number of cores used for running the forward step, not for init (which is what would happen here, and which isn't appropriate).

Reply:
Let me know if this is something you'd like help with.  I could make the commit to update `ForwardStep` if that would help you make progress on this PR.